examples/wgpu-triangle/main.rs

```diff
@@ -0,0 +1,270 @@
+            format: swapchain_format,
+            width: size.0,
+            height: size.1,
+            present_mode: wgpu::PresentMode::Fifo,
+            alpha_mode: surface.get_supported_alpha_modes(&adapter)[0],
+        };
+
+        surface.configure(&device, &config);
+
+        Self {
+            config,
+            surface,
+            window,
+            device,
+            render_pipeline,
+            queue,
+        }
+    }
+
+    #[cfg(target_os = "macos")]
+    fn schedule_render(&self) {
+        self.window
+            .get_idle_handle()
+            .unwrap()
+            .schedule_idle(IdleToken::new(0));
+    }
+
+    #[cfg(not(target_os = "macos"))]
+    fn schedule_render(&self) {
+        self.window.invalidate();
+    }
+
+    fn draw(&mut self) {
+        let frame = self
+            .surface
+            .get_current_texture()
+            .expect("Failed to acquire next swap chain texture");
+        let view = frame
+            .texture
+            .create_view(&wgpu::TextureViewDescriptor::default());
+        let mut encoder = self
+            .device
+            .create_command_encoder(&wgpu::CommandEncoderDescriptor { label: None });
+        {
+            let mut rpass = encoder.begin_render_pass(&wgpu::RenderPassDescriptor {
+                label: None,
+                color_attachments: &[Some(wgpu::RenderPassColorAttachment {
+                    view: &view,
+                    resolve_target: None,
+                    ops: wgpu::Operations {
+                        load: wgpu::LoadOp::Clear(wgpu::Color::GREEN),
+                        store: true,
+                    },
+                })],
+                depth_stencil_attachment: None,
+            });
+            rpass.set_pipeline(&self.render_pipeline);
+            rpass.draw(0..3, 0..1);
+        }
+
+        self.queue.submit(Some(encoder.finish()));
+        frame.present();
+    }
+}
+
+fn main() {
+    let app = Application::new().unwrap();
+    let mut window_builder = glazier::WindowBuilder::new(app.clone());
+    window_builder.resizable(true);
+    window_builder.set_size((WIDTH as f64 / 2., HEIGHT as f64 / 2.).into());
+    window_builder.set_handler(Box::new(WindowState::new()));
+    let window_handle = window_builder.build().unwrap();
+    window_handle.show();
+    app.run(None);
+}
+
+struct WindowState {
+    inner: Option<InnerWindowState>,
+}
+
+impl WindowState {
+    fn new() -> Self {
+        Self { inner: None }
+    }
+}
+
+impl WinHandler for WindowState {
+    fn connect(&mut self, handle: &WindowHandle) {
+        let inner = InnerWindowState::create(handle.clone());
+        inner.schedule_render();
+        self.inner = Some(inner);
+    }
+
+    fn prepare_paint(&mut self) {}
+
+    fn paint(&mut self, _: &Region) {
+        let inner = self.inner.as_mut().unwrap();
+        inner.draw();
+        inner.schedule_render();
+    }
+
+    fn idle(&mut self, _: IdleToken) {
```

Comment:
Maybe it's worth omitting the methods with default implementations, given that they aren't the focus of the example? Up to you...

Reply:
good idea! seems like `idle` was actually necessary, but removed the other ones